gluten-uniffle/velox/src/main/java/org/apache/spark/shuffle/writer/VeloxUniffleColumnarShuffleWriter.java

```diff
@@ -189,11 +189,13 @@ public long spill(MemoryTarget self, Spiller.Phase phase, long size) {
       }
     }
 
-    long startTime = System.nanoTime();
     LOG.info("nativeShuffleWriter value {}", nativeShuffleWriter);
-    if (nativeShuffleWriter == -1L) {
-      throw new IllegalStateException("nativeShuffleWriter should not be -1L");
+    // If all of the ColumnarBatch have empty rows, the nativeShuffleWriter still equals -1
+    if (nativeShuffleWriter == -1L && !isMemoryShuffleEnabled) {
+      super.sendCommit();
+      return;
     }
```

Comment:
```suggestion
    if (nativeShuffleWriter == -1L) {
      if (!isMemoryShuffleEnabled) {
        super.sendCommit();
      }
      return;
    }
```

Reply:
@xianjingfeng, when `!records.hasNext()` returns true but `isMemoryShuffleEnabled` is true, the `VeloxUniffleColumnarShuffleWriter#writeImpl` does not also return. This change keeps the same behavitor with above.

Reply:
> @xianjingfeng, when `!records.hasNext()` returns true but `isMemoryShuffleEnabled` is true, the `VeloxUniffleColumnarShuffleWriter#writeImpl` does not also return. This change keeps the same behavior with above.

I don't think the above logic makes sense either.

Reply:
https://github.com/apache/incubator-uniffle/blob/master/client-spark/spark3/src/main/java/org/apache/spark/shuffle/writer/RssShuffleWriter.java#L341-L343